### PR TITLE
fixed Unwanted column removal while dragging, Removed Unwanted backen…

### PR DESCRIPTION
**Pull request by 602dhruviii**
Description

1. Put the query text on the next line when the `|` character is entered 
Screenshot 2024-07-09 111036

2. Scroll on dashboards page is making multiple call to backed
Screenshot 2024-07-09 111149

3.Prevent Column Removal in Logs Table When Moving Header

4.Show Alert Rule Name at the Top when we are on edit mode.
Screenshot 2024-07-09 122402

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: static/js/dashboard-logs-results-grid.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ function getPanelGridOptions() {
                 if (diff <= 5) {
                     let scrollingTrigger = true;
                     data = getQueryParamsData(scrollingTrigger);
-                    runPanelLogsQuery(data);
+
```
Comment (sunitakawane) [inline static/js/dashboard-logs-results-grid.js:105]: this is needed to have scrolling data. If you remove runPanelLogsQuery we will always get fixed set of records without scrolling

File: static/js/dashboard.js [hunk 2]
```diff
@@ -28,6 +28,19 @@ let originalIndexValues = [];
 let indexValues = [];
 let isDefaultDashboard = false;
 $(document).ready(async function () {
+    $("#query-input-box").keydown(function (e) {
```
[the remark this answers is not in the document]
Comment (602dhruviii) [reply static/js/dashboard.js:31]: yeah like JIRA says that Put the query text on the next line when the `|` character is entered, so for that I have applied to textarea which takes query.

File: static/js/dashboard-logs-results-grid.js (hunk 1 above)
Comment (602dhruviii) [reply static/js/dashboard-logs-results-grid.js:105]: okay, would modify the function then.

File: static/alert.html [hunk 3]
```diff
@@ -104,6 +104,7 @@
                         <button class="btn" id="save-alert-btn" type="submit" tabindex="10">Save</button>
                     </div>
                     <div class="add-alert-form">
+                        <h1 class="rulename"></h1>
```
Comment (sonamgupta21) [inline static/alert.html:107]: Make the CSS for Rulename as 
```     
    font-size: 18px;
    font-weight: 700;
    margin-bottom: 20px;
    margin-left: 5px;
  ```
Also, set the display to none for the creation time and show it only during edit time.

File: static/js/dashboard.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sonamgupta21) [reply static/js/dashboard.js:31]: It should also work when a user pastes a query string containing `|`.

File: static/js/dashboard-logs-results-grid.js [hunk 4]
```diff
@@ -112,6 +112,10 @@ function getPanelGridOptions() {
 
 
 function renderPanelLogsGrid(columnOrder, hits, panelId,currentPanel) {
+    if (typeof panelId === 'undefined' || panelId === null) {
```
Comment (sonamgupta21) [inline static/js/dashboard-logs-results-grid.js:115]: image
Still shows an error when we scroll.

**Review by sonamgupta21 — APPROVED** [review]

**Pull request merged**